sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2128,38 +2128,77 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     }
   }
 
-  test("SPARK-23849: schema inferring touches less data if samplingRation < 1.0") {
-    val predefinedSample = Set[Int](2, 8, 15, 27, 30, 34, 35, 37, 44, 46,
+  val sampledTestData = (value: java.lang.Long) => {
```

Comment:
@MaxGekk, can we have the data in `TestJsonData`, for example,

```scala
def sampledTestData: Dataset[String] =
    spark.createDataset(spark.sparkContext.parallelize(
      ...
    )(Encoders.STRING)
```

and use it, for example, `sampledTestData.coalesce(1)` in `JsonSuite`?